### PR TITLE
SLVSCODE-32 Rework code actions and commands

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseError;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseErrorCode;
+import org.sonarsource.sonarlint.core.client.api.common.RuleDetails;
+import org.sonarsource.sonarlint.core.client.api.connected.ConnectedSonarLintEngine;
+import org.sonarsource.sonarlint.ls.SonarLintExtendedLanguageClient.ShowRuleDescriptionParams;
+import org.sonarsource.sonarlint.ls.connected.ProjectBindingManager;
+import org.sonarsource.sonarlint.ls.connected.ProjectBindingWrapper;
+
+import static java.net.URI.create;
+import static org.sonarsource.sonarlint.ls.AnalysisManager.SONARLINT_SOURCE;
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java]: Unused import?

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+public class CommandManager {
+
+  private static final String SONARLINT_OPEN_STANDALONE_RULE_DESCRIPTION_COMMAND = "SonarLint.OpenStandaloneRuleDesc";
+  static final String SONARLINT_OPEN_RULE_DESCRIPTION_FROM_CODE_ACTION_COMMAND = "SonarLint.OpenRuleDescCodeAction";
+  private static final String SONARLINT_DEACTIVATE_RULE_COMMAND = "SonarLint.DeactivateRule";
+  static final String SONARLINT_UPDATE_ALL_BINDINGS_COMMAND = "SonarLint.UpdateAllBindings";
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java]: I'd make all of these package-private, for consistency + readability.

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  static final List<String> SONARLINT_COMMANDS = Arrays.asList(
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java]: I believe this deserves to be wrapped with `Collections.unmodifiableList`.

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java]: No

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java (hunk 2 above)
Comment (henryju) [reply src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java]: :+1:

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java (hunk 3 above)
Comment (henryju) [reply src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java]: This add a cost for no gain IMO. This is useful when you return a collection in an API that should be immutable, to preserve from unexpected modifications by the caller. Here this is an internal constant.

**Pull request merged**